web/api/queries/query.c

```diff
@@ -403,116 +403,254 @@ static inline void do_dimension(
+        // make sure we return data in the proper time range
+        if (unlikely(now > before_wanted)) {
+#ifdef NETDATA_INTERNAL_CHECKS
+            r->internal.log = "stopped, because attempted to access the db after 'wanted before'";
+#endif
+            break;
+        }
+        if (unlikely(now < after_wanted)) {
+#ifdef NETDATA_INTERNAL_CHECKS
+            r->internal.log = "skipped, because attempted to access the db before 'wanted after'";
+#endif
+            continue;
+        }
+
+        while (now >= db_now && (!rd->state->query_ops.is_finished(&handle) ||
+                                 does_storage_number_exist(n_prev))) {
+            value = NAN;
+            if (does_storage_number_exist(n_prev)) {
+                // use the previously read database value
+                n_curr = n_prev;
+            } else {
+                // read the value from the database
+                n_curr = rd->state->query_ops.next_metric(&handle, &db_now);
+            }
+            n_prev = SN_EMPTY_SLOT;
+            // db_now has a different value than above
+            if (likely(now >= db_now)) {
+                if (likely(does_storage_number_exist(n_curr))) {
+                    value = unpack_storage_number(n_curr);
+                    if (likely(value != 0.0))
+                        values_in_group_non_zero++;
+
+                    if (unlikely(did_storage_number_reset(n_curr)))
+                        group_value_flags |= RRDR_VALUE_RESET;
+                }
+            } else {
+                // We must postpone processing the value and fill the result with gaps instead
+                if (likely(does_storage_number_exist(n_curr))) {
+                    n_prev = n_curr;
+                }
+            }
+            // add this value to grouping
+            r->internal.grouping_add(r, value);
+            values_in_group++;
+            db_points_read++;
+        }
+
+        if (0 == values_in_group) {
+            // add NAN to grouping
+            r->internal.grouping_add(r, NAN);
+        }
+
+        rrdr_line = rrdr_line_init(r, now, rrdr_line);
+
+        if(unlikely(!min_date)) min_date = now;
+        max_date = now;
+
+        // find the place to store our values
+        RRDR_VALUE_FLAGS *rrdr_value_options_ptr = &r->o[rrdr_line * r->d + dim_id_in_rrdr];
+
+        // update the dimension options
+        if(likely(values_in_group_non_zero))
+            r->od[dim_id_in_rrdr] |= RRDR_DIMENSION_NONZERO;
+
+        // store the specific point options
+        *rrdr_value_options_ptr = group_value_flags;
+
+        // store the value
+        value = r->internal.grouping_flush(r, rrdr_value_options_ptr);
+        r->v[rrdr_line * r->d + dim_id_in_rrdr] = value;
+
+        if(likely(points_added || dim_id_in_rrdr)) {
+            // find the min/max across all dimensions
+
+            if(unlikely(value < min)) min = value;
+            if(unlikely(value > max)) max = value;
```

Comment:
I'm not sure there's any point in using `unlikely` here.

Reply:
It's just Costa's copy-pasta code I moved around. Just following the logic "if it ain't broke don't fix it".